contributingGuides/STYLE.md

```diff
@@ -10,13 +69,514 @@ We use Prettier to automatically style our code.
+
+```ts
+// BAD
+const sizingStyles = {
+    w50: {
+        width: '50%',
+    },
+    mw100: {
+        maxWidth: '100%',
+    },
+} as const;
+
+// GOOD
+const sizingStyles = {
+    w50: {
+        width: '50%',
+    },
+    mw100: {
+        maxWidth: '100%',
+    },
+} satisfies Record<string, ViewStyle>;
+```
+
+### Type imports/exports
+
+Always use the `type` keyword when importing/exporting types
+
+> Why? In order to improve code clarity and consistency and reduce bundle size after typescript transpilation, we enforce the all type imports/exports to contain the `type` keyword. This way, TypeScript can automatically remove those imports from the transpiled JavaScript bundle 
+
+Imports:
+```ts
+// BAD
+import {SomeType} from './a'
+import someVariable from './a'
+
+import {someVariable, SomeOtherType} from './b'
+
+// GOOD
+import type {SomeType} from './a'
+import someVariable from './a'
+```
+
+ Exports:
+```ts
+// BAD
+export {SomeType}
+export someVariable
+// or 
+export {someVariable, SomeOtherType}
+
+// GOOD
+export type {SomeType}
+export someVariable
+```
+
+### Exception to Rules
+
+Most of the rules are enforced in ESLint or checked by TypeScript. If you think your particular situation warrants an exception, post the context in the `#expensify-open-source` Slack channel with your message prefixed with `TS EXCEPTION:`. The internal engineer assigned to the PR should be the one that approves each exception, however all discussion regarding granting exceptions should happen in the public channel instead of the GitHub PR page so that the TS migration team can access them easily.
+
+When an exception is granted, link the relevant Slack conversation in your PR. Suppress ESLint or TypeScript warnings/errors with comments if necessary.
+
+This rule will apply until the migration is done. After the migration, discussion on granting exception can happen inside the PR page and doesn't need take place in the Slack channel.
+
+### Communication Items
+
+> Comment in the `#expensify-open-source` Slack channel if any of the following situations are encountered. Each comment should be prefixed with `TS ATTENTION:`. Internal engineers will access each situation and prescribe solutions to each case. Internal engineers should refer to general solutions to each situation that follows each list item.
+
+- I think types definitions in a third party library or JavaScript's built-in module are incomplete or incorrect
+
+When the library indeed contains incorrect or missing type definitions and it cannot be updated, use module augmentation to correct them. All module augmentation code should be contained in `/src/types/modules/*.d.ts`, each library as a separate file.
+
+```ts
+// external-library-name.d.ts
+
+declare module "external-library-name" {
+    interface LibraryComponentProps {
+        // Add or modify typings
+        additionalProp: string;
+    }
+}
+```
```

Comment:
Same here, I don't think we need it. The only message with "TS ATTENTION" from slack was form Hayata [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1691772060882829) 😄